packages/component-library/src/components/form/mt-button/mt-button.vue

```diff
@@ -21,133 +21,60 @@
-      default: false,
-    },
-    /**
-     * Change the look of the button
-     * Values: primary, secondary, critical, action
-     * @values primary, secondary, critical, action
-     */
-    variant: {
-      type: String as PropType<"primary" | "secondary" | "critical" | "action">,
-      required: false,
-      default: "",
-      validator(value: string) {
-        if (!value.length) {
-          return true;
-        }
-        return ["primary", "secondary", "critical", "action"].includes(value);
-      },
-    },
-    ghost: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    /**
-     * Change the size of the button
-     * @values x-small, small, default, large
-     */
-    size: {
-      type: String,
-      required: false,
-      default: "small",
-      validator(value: string) {
-        if (!value.length) {
-          return true;
-        }
-        return ["x-small", "small", "default", "large"].includes(value);
-      },
-    },
-    /**
-     * The button will be rendered as a square. You need to consider the text length
-     * if you activate this property.
-     */
-    square: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    /**
-     * Renders the button as a block element instead of an inline-block element. The button
-     * fills up all horizontal space.
-     */
-    block: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    /**
-     * If a link is provided then the user gets redirected to a new tab on a click.
-     */
-    link: {
-      type: String,
-      required: false,
-      default: null,
-    },
-    /**
-     * Shows a loading indicator instead of the text.
-     */
-    isLoading: {
-      type: Boolean,
-      default: false,
-      required: false,
-    },
-  },
-
-  computed: {
-    buttonClasses() {
-      return {
-        [`mt-button--${this.variant}${this.allowGhostVariant ? "-ghost" : ""}`]: !!this.variant,
-        [`mt-button--${this.size}`]: !!this.size,
-        "mt-button--block": this.block,
-        "mt-button--disabled": this.disabled,
-        "mt-button--square": this.square,
-      };
-    },
-
-    allowGhostVariant() {
-      return this.ghost && this.variant !== "secondary";
-    },
-
-    contentVisibilityClass() {
-      return {
-        "mt-button__content--hidden": this.isLoading,
-      };
-    },
-
-    iconSize() {
-      return this.size === "x-small" ? 8 : this.size === "large" ? 12 : 10;
-    },
+import { computed } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    disabled?: boolean;
+    variant?: "primary" | "secondary" | "critical" | "action";
+    ghost?: boolean;
+    size?: "x-small" | "small" | "default" | "large";
+    square?: boolean;
+    block?: boolean;
+    link?: string;
+    isLoading?: boolean;
+  }>(),
+  {
+    // @ts-expect-error -- This is to keep the default behaviour
+    variant: "",
+    default: "default",
```

Comment:
@Haberkamp wouldn't `size` need to be `"default"`? But maybe I'm just not familiar with the `default` property